src/python_testing/TC_SC_4_3.py

```diff
@@ -0,0 +1,262 @@
+#
+#    Copyright (c) 2024 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+import re
+
+import chip.clusters as Clusters
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mdns_discovery.mdns_discovery import MdnsDiscovery, MdnsServiceType
+from mobly import asserts
+
+'''
+Category
+Functional conformance
+
+Purpose
+The purpose of this test case is to verify that a Matter node is discoverable
+and can advertise its services in a Matter network.
+
+Test Plan
+https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/securechannel.adoc#343-tc-sc-43-discovery-dut_commissionee
+'''
+
+
+class TC_SC_4_3(MatterBaseTest):
+
+    ONE_HOUR_IN_MS = 3600000
+    MAX_SAT_VALUE = 65535
+    MAX_T_VALUE = 6
+
+    async def get_descriptor_server_list(self):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=self.default_controller,
+            cluster=Clusters.Descriptor,
+            attribute=Clusters.Descriptor.Attributes.ServerList
+        )
+
+    async def get_idle_mode_threshhold_ms(self):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=self.default_controller,
+            cluster=Clusters.IcdManagement,
+            attribute=Clusters.IcdManagement.Attributes.ActiveModeThreshold
+        )
+
+    async def get_icd_feature_map(self):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=self.default_controller,
+            cluster=Clusters.IcdManagement,
+            attribute=Clusters.IcdManagement.Attributes.FeatureMap
+        )
+
+    def get_dut_instance_name(self) -> str:
+        node_id = self.dut_node_id
+        compressed_fabric_id = self.default_controller.GetCompressedFabricId()
+        instance_name = f'{compressed_fabric_id:016X}-{node_id:016X}'
+        return instance_name
+
+    def get_operational_subtype(self) -> str:
+        compressed_fabric_id = self.default_controller.GetCompressedFabricId()
+        service_name = f'_I{compressed_fabric_id:016X}._sub.{MdnsServiceType.OPERATIONAL.value}'
+        return service_name
+
+    @staticmethod
+    def verify_decimal_value(input_value, comparison_value: int):
```

Comment:
this seems always used followed by an assert. Could make non-static and include the assert in it. Like `AssertValidDecimalValue` or something like that.

Reply:
I set it up this way so I get a success boolean back which will pass the assert, and when False, it also returns the specific reason.

I think the assert looks better in the test step code block as it unmistakenly indicates that a specific verification is there and improves readability.

Please advise if I misunderstood the comment.